### PR TITLE
go 1.17 upgrade for prow images

### PR DESCRIPTION
**Pull request by vijtrip2**
Description of changes:

* include updating go module version inside `go.mod` file alongwith updating ACK runtime version
* perform `go mod download` and `go mod tidy` before executing `make build-controller` command
* update `go` version to `1.17.5` in all prow images
* use github utility functions for opening PR and issues inside `auto-generate-controllers.sh`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cd/auto-generate/auto-generate-controllers.sh [hunk 1]
```diff
@@ -79,6 +82,14 @@ else
   echo "auto-generate-controllers.sh][INFO] ACK runtime version for new controllers will be $ACK_RUNTIME_VERSION"
 fi
 
+GO_MODULE_VERSION=$(grep -E "^go [0-9]+\.[0-9]+$" go.mod | cut -d " " -f2)
```
Comment (jaypipes) [inline cd/auto-generate/auto-generate-controllers.sh:85]: ah, you are referring to the Go *language* version, not the Go *module* version, right?

Comment (jaypipes) [reply cd/auto-generate/auto-generate-controllers.sh:85]: In other words, this variable name should be `GO_VERSION` or `GO_LANG_VERSION`, yes?

File: cd/auto-generate/auto-generate-controllers.sh [hunk 2]
```diff
@@ -251,44 +246,15 @@ for CONTROLLER_NAME in $CONTROLLER_NAMES; do
...
-      echo "ok"
-    fi
-
     # Capture 'make build-controller' command output, then persist
     # in '$GITHUB_PR_BODY_FILE'
     MAKE_BUILD_OUTPUT=$(cat "$MAKE_BUILD_OUTPUT_FILE")
     PR_BODY_TEMPLATE_FILE_NAME=$([[ -z "$RELEASE_VERSION" ]] && echo "gh_pr_body_template.txt" || echo "gh_pr_body_new_release_template.txt")
     GITHUB_PR_BODY_TEMPLATE_FILE="$THIS_DIR/$PR_BODY_TEMPLATE_FILE_NAME"
-    GITHUB_PR_BODY_FILE=/tmp/"SERVICE_NAME"_gh_pr_body
+    GITHUB_PR_BODY_FILE=/tmp/"$SERVICE_NAME"_gh_pr_body
```
Comment (jaypipes) [inline cd/auto-generate/auto-generate-controllers.sh:254]: heh, nice catch! :)

File: cd/auto-generate/auto-generate-controllers.sh (hunk 1 above)
Comment (vijtrip2) [reply cd/auto-generate/auto-generate-controllers.sh:85]: yes, I will update the variable name.

File: cd/auto-generate/auto-generate-controllers.sh [hunk 3]
```diff
@@ -79,6 +82,14 @@ else
   echo "auto-generate-controllers.sh][INFO] ACK runtime version for new controllers will be $ACK_RUNTIME_VERSION"
 fi
 
+GO_VERSION_IN_GO_MOD=$(grep -E "^go [0-9]+\.[0-9]+$" go.mod | cut -d " " -f2)
+if [[ -z $GO_VERSION_IN_GO_MOD ]]; then
+  echo "auto-generate-controllers.sh][ERROR] Unable to determine go module version from code-generator/go.mod file. Exiting"
+  exit 1
+else
+  echo "auto-generate-controllers.sh][INFO] go module version in code-generator/go.mod file is $GO_VERSION_IN_GO_MOD"
```
Comment (jaypipes) [inline cd/auto-generate/auto-generate-controllers.sh:90]: Can you change this to " go version" instead of "go module version"?

**Review by RedbackThomson — APPROVED** [review]
/lgtm

**Pull request merged**